src/formatDate.js

```diff
@@ -8,7 +8,46 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+
+  const dateInOldFormatArr = date.split(oldSeparator);
+  const dateInNewFormatArr = [];
+
+  const month = dateInOldFormatArr[fromFormat.indexOf('MM')];
+  const day = dateInOldFormatArr[fromFormat.indexOf('DD')];
+  let year;
+
+  if (date.length === 10) {
+    year = dateInOldFormatArr[fromFormat.indexOf('YYYY')];
+    year = `${year[2]}${year[3]}`;
+  } else if (date.length === 8) {
+    year = dateInOldFormatArr[fromFormat.indexOf('YY')];
+  }
```

Comment:
The condition to determine the year format based on the length of the date string is not reliable. Different date formats can result in the same length. For instance, 'DD-MM-YY' and 'MM-DD-YYYY' are both 8 characters long but have different year formats. Instead, use the `fromFormat` array to determine if the year is in 'YYYY' or 'YY' format.